logging-account-cloudtrail.yaml

```diff
@@ -235,24 +318,28 @@ Resources:
                 kms:CallerAccount: !Ref 'AWS::AccountId'
   CloudTrailEncryptionKeyAlias:
     Type: AWS::KMS::Alias
+    Condition: ShouldEncryptLogs
     Properties:
       AliasName: !Sub 'alias/${KMSKeyName}'
       TargetKeyId: !Ref 'CloudTrailEncryptionKey'
 
   LogsAccountCloudTrail:
     Type: AWS::CloudTrail::Trail
+    Condition: LocalCloudTrail
     Properties:
-      EnableLogFileValidation: true
-      IncludeGlobalServiceEvents: true
+      EnableLogFileValidation: !Ref EnableLogFileValidation
+      IncludeGlobalServiceEvents: !Ref IncludeGlobalEvents
       IsLogging: true
-      IsMultiRegionTrail: true
-      S3BucketName: !Ref 'CloudTrailBucketName'
-      KMSKeyId: !Ref 'CloudTrailEncryptionKey'
+      IsMultiRegionTrail: !Ref MultiRegion
+      EventSelectors: 
+        - ReadWriteType: !Ref ReadWriteType
+      S3BucketName: !Ref CloudTrailBucketName
+      KMSKeyId: !If [ ShouldEncryptLogs, !GetAtt CloudTrailEncryptionKey.Arn, !Ref "AWS::NoValue" ] 
       TrailName: !Ref CloudTrailLogName
 Outputs:
   CloudTrailEncryptionKeyArn:
```

Comment:
Use a `Condition: ShouldEncryptLogs` key here instead to either add the Output or not. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/conditions-section-structure.html

Reply:
Good catch! I forgot about Conditions for Outputs. It's been added now.